product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -181,16 +184,18 @@ def _select_customerinfo(
         precision = self.env["decimal.precision"].precision_get(
             "Product Unit of Measure"
         )
-        res = self._customers_filter_by_quantity(
-            customers, quantity=quantity, uom_id=uom_id, precision=precision
-        )
-        if res:
-            customer_name = res[0].name
-            res = res.filtered(lambda x, name=customer_name: x.name == name)
+
+        if quantity is not None:
+            customers = self._customers_filter_by_quantity(
```

Comment:
Wrong variable names lead to later confusions:

```suggestion
            customerinfos = self._customers_filter_by_quantity(
```

and instead of `customer_name` put simply `customer`, as it represents a res.partner.

Reply:
We'll make this change in another PR